lib/components/note-preview/index.tsx

```diff
@@ -29,12 +31,46 @@ type Props = OwnProps & StateProps & DispatchProps;
 export const NotePreview: FunctionComponent<Props> = ({
   editNote,
   fontSize,
+  isDialogOpen,
+  isNoteInfoOpen,
   note,
   noteId,
   searchQuery,
 }) => {
   const previewNode = useRef<HTMLDivElement>();
 
+  useEffect(() => {
+    const copyRenderedNote = (event: ClipboardEvent) => {
+      console.log(isNoteInfoOpen);
+      // TODO this isn't working - these are false when I log them, but in console they are true
+      // e.g. state.ui.showNoteInfo in console will print "true"
+      // Only copy if not viewing the note info panel or a dialog
+      if (isNoteInfoOpen || isDialogOpen) {
+        return true;
+      }
+
+      // Only copy the rendered content if nothing is selected
+      if (!document.getSelection().isCollapsed) {
+        return true;
+      }
+
+      const node = document.createDocumentFragment();
+      const div = document.createElement('div');
+      renderToNode(div, note!.content, searchQuery);
```

Comment:
nitpick and personal style: I'm more in favor of handling these possible-null errors in one of two other ways than to use the `!`. why? `!` tells TypeScript to stop checking and leaves it a fully-manual operation meaning that (a) we can get it wrong and (b) if things change we may not realize it.

If all we need is a safe fallback I have so-far preferred using optional-chaining and the null-coalescing operator. It's explicit what we're falling back to…

```ts
renderToNode(div, note?.content ?? '', searchQuery );
```

…or if we "know" that it's there even though the types and TypeScript show that there's a possibility that it might not be, then I actually prefer leaving the error in the code since it can make it easy for us to find later. there's a decent amount of future reworking we could do in this app to eliminate these case. basically anywhere we have components rendering that depend on the opened or selected note we're going to have these errors.

as far as I can tell right now this may never happen because it would be a hefty change to do that for relatively little gain. we could pass a copy of the `note` down from a parent component instead of sourcing it from state: this would eliminate the possibility of a null/undefined value but introduce duplication of data and reintroduce prop-drilling.

Reply:
I copied this from https://github.com/Automattic/simplenote-electron/blob/rewrite/no-races/lib/components/note-preview/index.tsx#L87 fwiw. I'm happy to change it if you think we ought.

Reply:
okay lol sorry. that's me being lazy and having a different "writing code" vs. "reviewing code" mindset

if I were writing it again I'd probably leave out the `!` for the reasons mentioned above. it's not a big deal but as I go I get more persuaded of the value of not overriding TypeScript's inference.

can do it later if you want outside of this PR

Reply:
oh I removed it and TS didn't even warn me it could be null so I guess we're good